mimic/imimic.py

```diff
@@ -28,13 +28,37 @@ def resource_for_region(region, uri_prefix,
         """
 
 
+class IEndpointTemplate(Interface):
+    """
+    AN :obj:`IEndpointTemplate` provides an interface for the OS-KSCATALOG
+    Admin endpoint templates, defining the fields.
+    """
+    id_key = Attribute("Unique Service ID")
+    region_key = Attribute("Region the Endpoint services")
+    type_key = Attribute("Service type, example: object-store")
+    name_key = Attribute("Service Name, example: Cloud Files")
+    enabled_key = Attribute("Whether the Endpoint is globally available")
+    public_url = Attribute("Internet Facing URL")
+    internal_url = Attribute("Intranet Facing URL")
+    admin_url = Attribute("Service Adminsitration URL")
+    tenant_alias = Attribute("Tenant Alias to be appended to the "
+                             "endpoint URI")
+    version_id = Attribute("Version Details")
+    version_info = Attribute("URL to get the version information")
+    version_list = Attribute("URL to determine which version is hosted "
+                             "by the endpoint")
+
+
 class IExternalAPIMock(Interface):
     """
     An :obj:`IExternalAPIMock` provides an API via an entry for services
     external to Mimic in a tenant's service catalog.
     """
+    id_key = Attribute("Unique Service ID")
     name_key = Attribute("Name of the External API. Primarily used for "
                          "the internal mappings. Must be unique.")
+    type_key = Attribute("Type of the Templatized External API. Primarily "
+                         "used for the internal mappings. Must be unique.")
```

Comment:
@glyph I should be the only one using these interfaces, `IExternalAPIMock` was something I introduced in a previous PR (#608) to differentiate the services this PR (and the next) will be managing versus the `IAPIMock` that has always been there. This was a refinement of the previous PR as the functionality matured. All of the `IExternalAPIMock` functionality will be done using the templating functionality (next PR will add the external RESTful API to support).